pkg/scale/encode.go

```diff
@@ -284,22 +382,46 @@ func (es *encodeState) encodeFixedWidthInt(i interface{}) (err error) {
 	switch i := i.(type) {
 	case int8:
 		err = binary.Write(es, binary.LittleEndian, byte(i))
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case uint8:
 		err = binary.Write(es, binary.LittleEndian, i)
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case int16:
 		err = binary.Write(es, binary.LittleEndian, uint16(i))
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case uint16:
 		err = binary.Write(es, binary.LittleEndian, i)
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case int32:
 		err = binary.Write(es, binary.LittleEndian, uint32(i))
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case uint32:
 		err = binary.Write(es, binary.LittleEndian, i)
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case int64:
 		err = binary.Write(es, binary.LittleEndian, uint64(i))
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	case uint64:
 		err = binary.Write(es, binary.LittleEndian, i)
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
+		}
 	default:
-		err = fmt.Errorf("could not encode fixed width integer, invalid type: %T", i)
+		err = fmt.Errorf("%w: invalid type: %T", ErrEncodeFixedWidthInt, i)
```

Comment:
maybe change this one to something like `ErrEncodeInvalidFixedWidthInt`?

Reply:
We should definitely panic here (it would be a 100% programming error from the scale package), so get rid of the sentinel error and just panic

```suggestion
		panic(fmt.Sprintf("invalid type: %T", i)
```